microsoft/PowerToys
# Move plugin specific options under pluigin settings

## Summary of the Pull Request

**What is this about:**
![image](https://user-images.githubusercontent.com/17161067/108510465-39e06e00-72c7-11eb-9038-c8c61620feac.png)


**What is include in the PR:** 
- Added `AdditionalOptions` property to `ISettingProvider`. List of options(checkboxes).
- Render plugin specific options in plugin settings
- Move disable drive detection warning to the indexer settings

**How does someone test / validate:** 
- Delete `PowerToys Run\settings.json`
- Check that `Disable drive detection warning for the file search plugin` is under Indexer settings

I just noticed that the indexer plugin has the wrong name. It is probably an issue with resources not in the scope of this issue.

## Quality Checklist

- [X] **Linked issue:** #9655

## Contributor License Agreement (CLA)
A CLA must be signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/PowerToys) and sign the CLA.
